test/connectors/hip/CalculationListLegacyConnectorSpec.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.hip
+
+import constants.BaseTestConstants.{testNino, testTaxYearEnd}
+import constants.CalculationListTestConstants._
+import mocks.MockHttpV2
+import models.hip.{ErrorResponse, GetLegacyCalcListApiName, UnexpectedJsonResponse}
+import uk.gov.hmrc.http.HttpResponse
+import utils.TestSupport
+
+class CalculationListLegacyConnectorSpec extends TestSupport with MockHttpV2 {
+
+  object TestCalculationListLegacyConnector extends CalculationListLegacyConnector(mockHttpClientV2, microserviceAppConfig)
+  val platform: String = microserviceAppConfig.hipUrl
+  val url1404 = s"$platform/itsd/calculations/liability/$testNino?taxYear=$testTaxYearEnd"
```

Comment:
can this be changed to:   
```
val url1404 = s"http://localhost:9084/itsd/calculations/liability/$testNino?taxYear=$testTaxYearEnd"
```
so it breaks if people change the config locally accidentally or for whatever reason? 
and then 

` val platform: String = microserviceAppConfig.hipUrl`

can be deleted I think.

Reply:
My point of view is, URL changes should be dynamic as in changes to config should not break tests, as we are not testing urls. Eg: if we need to change our calculation apis to point to different stub then this should not be failing localtests.
Also this is same behaviour with all connector tests i.e., they are injecting config. So will leave it for now if thats ok and do it as an improvements for all tests please.
Also, ATs fail anyways if those changes are accidently changed.

Reply:
okay cool thanks for explaining